api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillEsRepo.kt

```diff
@@ -62,6 +71,32 @@ class CustomRichSkillQueriesImpl @Autowired constructor(override val elasticSear
     CustomRichSkillQueries {
     override val javaClass = RichSkillDoc::class.java
 
+    override fun getUuidsFromApiSearch(
+        apiSearch: ApiSearch,
+        statuses: Array<String>,
+        pageable: Pageable,
+        user: Jwt?,
+        collectionId: String?
+    ): List<String> {
+        val publishStatuses = statuses.mapNotNull {
+            val status = PublishStatus.forApiValue(it)
+            if (user == null && (status == PublishStatus.Deleted  || status == PublishStatus.Draft)) null else status
+        }.toSet()
+        var uuids: List<String> = emptyList()
```

Comment:
this can be a val and actually it would be more safe to use mapNotNull instead of map so also you can get rid of the .toList() and the 'as' cast: 

```
val uuids: List<String> = if (apiSearch.uuids != null)
        {
            apiSearch.uuids
        }
        else {
            val searchHits = byApiSearch(
                apiSearch,
                publishStatuses,
                Pageable.unpaged(),
                StringUtils.EMPTY
            )
            searchHits.mapNotNull { it.id }
        }
        return uuids
```